x-pack/plugins/secops/public/components/page/hosts/events_table/index.tsx

```diff
@@ -80,37 +144,43 @@ const getEventsColumns = (startDate: number, endDate: number) => [
     sortable: true,
     truncateText: true,
     hideForMobile: true,
-    render: (item: EventItem) => (
-      <ProviderContainer>{getOr('--', 'event.type', item)}</ProviderContainer>
-    ),
+    render: ({ event }: { event: Ecs }) => <>{getOr('--', 'event.type', event)}</>,
   },
   {
     name: 'Source',
     truncateText: true,
-    render: (item: EventItem) => (
+    render: ({ event }: { event: Ecs }) => (
       <>
-        {getOr('--', 'source.ip', item).slice(0, 12)} : {getOr('--', 'source.port', item)}
+        {formatSafely('source.ip', event)} : {getOr('--', 'source.port', event)}
       </>
     ),
   },
   {
     name: 'Destination',
     sortable: true,
     truncateText: true,
-    render: (item: EventItem) => (
+    render: ({ event }: { event: Ecs }) => (
       <>
-        {getOr('--', 'destination.ip', item).slice(0, 12)} : {getOr('--', 'destination.port', item)}
+        {formatSafely('destination.ip', event)} : {getOr('--', 'destination.port', event)}
       </>
     ),
   },
   {
     name: 'Location',
     sortable: true,
     truncateText: true,
-    render: (item: EventItem) => (
+    render: ({ event }: { event: Ecs }) => (
       <>
-        {getOr('--', 'geo.region_name', item)} - {getOr('--', 'geo.country_iso_code', item)}
+        {getOr('--', 'geo.region_name', event)} : {getOr('--', 'geo.country_iso_code', event)}
       </>
     ),
   },
 ];
+
+export const formatSafely = (path: string, data: Ecs) => {
+  if (has(path, data)) {
+    const txt = getOr('--', path, data);
+    return txt && txt.slice ? txt.slice(0, 12) : txt;
```

Comment:
I haven't seen checks for functions really before such as `txt.slice`. Is that a way of testing the type of txt to ensure it's an array or object that has a property or function called slice on it?

Optional, I would just use `isArray` from lodash

Reply:
This should go away when we have real data and not fake data